File: skills/video-prompt-generator/templates.py
"""
Video Prompt Templates - 风格模板系统

提供多种视频风格的提示词模板和增强技巧
"""

import random
from typing import Dict, List, Optional

# ====== 视频风格模板 ======

VIDEO_STYLES: Dict[str, Dict] = {
    "landscape": {
        "name": "风景风光",
        "name_en": "Landscape & Scenery",
        "description": "自然风景、城市风光、季节变化",
        "keywords": [
            "natural scenery", "mountains", "ocean", "forest", "sunset", "skyline",
            "urban cityscape", "countryside", "waterfall", "snow landscape",
            "自然风景", "山川", "海洋", "森林", "日落", "城市天际线",
            "田园风光", "瀑布", "雪景", "季节变化"
        ],
        "enhancements": {
            "lighting": ["golden hour", "sunrise", "sunset", "diffused light", "natural light"],
            "composition": ["wide angle", "panoramic", "aerial view", "landscape orientation"],
            "mood": ["serene", "majestic", "peaceful", "breathtaking", "epic"],
            "technical": ["4K", "8K", "hyper-detailed", "slow motion", "time-lapse"]
        }
    },
    "product": {
        "name": "产品展示",
        "name_en": "Product Showcase",
        "description": "产品特写、使用场景、电商广告",
        "keywords": [
            "product close-up", "360 degree rotation", "product demonstration",
            "lifestyle shot", "e-commerce", "commercial",
            "产品特写", "360度展示", "产品演示", "生活场景", "电商广告", "商业"
        ],
        "enhancements": {
            "lighting": ["studio lighting", "soft box lighting", "ring light", "professional lighting"],
            "composition": ["close-up", "macro shot", "isometric view", "white background"],
            "mood": ["professional", "clean", "elegant", "trustworthy", "premium"],
            "technical": ["4K", "sharp focus", "depth of field", "smooth camera movement"]
        }
    },
    "tech": {
        "name": "科技未来",
        "name_en": "Tech & Future",
        "description": "赛博朋克、未来城市、AI 主题",
        "keywords": [
            "cyberpunk", "futuristic city", "AI technology", "neon lights",
            "hologram", "digital interface", "tech innovation", "robotics",
            "赛博朋克", "未来城市", "AI技术", "霓虹灯", "全息投影",
            "数字界面", "科技创新", "机器人"
        ],
        "enhancements": {
            "lighting": ["neon lights", "cyberpunk colors", "blue and purple", "glowing", "dynamic lighting"],
            "composition": ["dynamic camera", "futuristic perspective", "digital overlay", "HUD elements"],
            "mood": ["futuristic", "innovative", "cutting-edge", "sleek", "high-tech"],
            "technical": ["4K", "digital effects", "particle effects", "motion graphics"]
        }
    },
    "emotional": {
        "name": "情感故事",
        "name_en": "Emotional Story",
        "description": "浪漫、怀旧、励志、感人",
        "keywords": [
            "romantic", "nostalgic", "inspirational", "heartwarming",
            "love story", "memory", "hope", "touching moments",
            "浪漫", "怀旧", "励志", "感人", "爱情故事", "回忆",
            "希望", "感人瞬间"
        ],
        "enhancements": {
            "lighting": ["warm lighting", "soft candlelight", "dramatic shadows", "cinematic lighting"],
            "composition": ["close-up emotions", "eye-level shot", "intimate framing", "shallow depth of field"],
            "mood": ["emotional", "touching", "nostalgic", "heartwarming", "inspiring"],
            "technical": ["cinematic", "slow motion", "film grain", "color grading"]
        }
    },
    "urban": {
        "name": "都市生活",
        "name_en": "Urban Life",
        "description": "街头、办公、咖啡店、都市夜景",
        "keywords": [
            "street photography", "office life", "cafe scene", "city night",
            "urban lifestyle", "modern architecture", "busy streets",
            "街头摄影", "办公生活", "咖啡店场景", "城市夜景",
            "都市生活", "现代建筑", "繁忙街道"
        ],
        "enhancements": {
            "lighting": ["city lights", "neon signs", "street lamps", "indoor lighting", "natural daylight"],
            "composition": ["street level view", "candid moments", "urban architecture", "lifestyle framing"],
            "mood": ["modern", "dynamic", "energetic", "authentic", "urban"],
            "technical": ["4K", "handheld camera", "natural movement", "documentary style"]
        }
    },
    "food": {
        "name": "美食烹饪",
        "name_en": "Food & Cooking",
        "description": "食物拍摄、烹饪过程、美食特写",
        "keywords": [
            "food photography", "cooking process", "delicious dish",
            "ingredients", "steaming food", "gourmet meal", "appetizing",
            "美食摄影", "烹饪过程", "美味佳肴", "食材", "热气腾腾",
            "美食", "诱人"
        ],
        "enhancements": {
            "lighting": ["appetizing lighting", "backlight for steam", "soft diffuse light", "warm food lighting"],
            "composition": ["close-up details", "overhead view", "45-degree angle", "plating showcase"],
            "mood": ["appetizing", "delicious", "mouth-watering", "cozy", "gourmet"],
            "technical": ["4K", "macro shots", "slow motion cooking", "steam effects"]
        }
    },
    "sports": {
        "name": "运动健身",
        "name_en": "Sports & Fitness",
        "description": "运动场景、健身日常、体育赛事",
        "keywords": [
            "sports action", "fitness training", "gym workout",
            "outdoor activities", "team sports", "athlete", "exercise",
            "运动动作", "健身训练", "健身房锻炼", "户外活动",
            "团队运动", "运动员", "运动"
        ],
        "enhancements": {
            "lighting": ["dynamic lighting", "stadium lights", "outdoor natural light", "action lighting"],
            "composition": ["action shots", "wide angle", "freeze frame", "motion blur"],
            "mood": ["energetic", "powerful", "dynamic", "inspiring", "competitive"],
            "technical": ["4K", "slow motion", "fast shutter", "action camera"]
        }
    },
    "traditional": {
        "name": "古风传统",
        "name_en": "Traditional Chinese",
        "description": "中国风、汉服、古装、古建筑",
        "keywords": [
            "Chinese traditional", "Hanfu", "ancient architecture",
            "classical scenery", "cultural heritage", "Chinese art",
            "中国风", "汉服", "古装", "古建筑", "古典风景",
            "文化遗产", "中国画风格"
        ],
        "enhancements": {
            "lighting": ["soft natural light", "candlelight", "lantern light", "golden hour"],
            "composition": ["classical framing", "symmetrical composition", "traditional perspective", "negative space"],
            "mood": ["elegant", "timeless", "cultured", "poetic", "graceful"],
            "technical": ["4K", "soft colors", "traditional aesthetics", "film-like quality"]
        }
    },
    "anime": {
        "name": "动漫二次",
        "name_en": "Anime & 2D Style",
        "description": "动漫风格、Q版、二次元",
        "keywords": [
            "anime style", "2D animation", "chibi", "manga",
            "Japanese animation", "colorful", "cute characters",
            "动漫风格", "二次元", "Q版", "漫画风格", "日式动画",
            "色彩丰富", "可爱角色"
        ],
        "enhancements": {
            "lighting": ["vibrant colors", "soft anime lighting", "glowing effects", "pastel tones"],
            "composition": ["dynamic poses", "character close-ups", "expressive angles", "anime framing"],
            "mood": ["energetic", "cute", "playful", "dreamy", "expressive"],
            "technical": ["2D style", "cel-shaded", "vibrant colors", "animated quality"]
        }
    },
    "abstract": {
        "name": "抽象艺术",
        "name_en": "Abstract Art",
        "description": "抽象、艺术、创意、实验性",
        "keywords": [
            "abstract art", "creative visuals", "experimental",
            "artistic expression", "unique perspectives", "surreal",
            "抽象艺术", "创意视觉", "实验性", "艺术表达",
            "独特视角", "超现实主义"
        ],
        "enhancements": {
            "lighting": ["dramatic lighting", "colored lighting", "silhouette", "experimental light"],
            "composition": ["abstract composition", "unconventional angles", "negative space", "geometric patterns"],
            "mood": ["artistic", "creative", "mysterious", "thought-provoking", "experimental"],
            "technical": ["4K", "color manipulation", "texture effects", "visual experiments"]
        }
    }
}

# ====== 提示词增强词库 ======

ENHANCEMENTS = {
    "lighting": {
        "golden hour": "黄金时刻的温暖光线",
        "soft lighting": "柔和的散射光",
        "dramatic shadows": "戏剧性的阴影",
        "neon lights": "霓虹灯光",
        "studio lighting": "专业影棚灯光",
        "natural light": "自然光",
        "sunrise": "日出",
        "sunset": "日落",
        "diffused light": "漫射光",
        "backlighting": "逆光",
        "rim light": "轮廓光",
        "ambient light": "环境光"
    },
    "composition": {
        "wide angle": "广角镜头",
        "close-up": "特写镜头",
        "aerial view": "鸟瞰视角",
        "rule of thirds": "三分法构图",
        "low angle": "低角度",
        "high angle": "高角度",
        "dutch angle": "荷兰角",
        "pan shot": "摇镜头",
        "tilt shot": "俯仰镜头",
        "tracking shot": "跟拍镜头",
        "360 degree": "360度全景",
        "macro shot": "微距拍摄"
    },
    "atmosphere": {
        "cinematic": "电影质感",
        "dreamy": "梦幻",
        "energetic": "充满活力",
        "mysterious": "神秘",
        "romantic": "浪漫",
        "nostalgic": "怀旧",
        "peaceful": "宁静",
        "dramatic": "戏剧性",
        "ethereal": "空灵",
        "urban": "都市感",
        "cozy": "温馨",
        "intense": "激烈"
    },
    "technical": {
        "4K": "4K超高清",
        "8K": "8K超高清",
        "slow motion": "慢动作",
        "time-lapse": "延时摄影",
        "hyper-detailed": "超精细细节",
        "film grain": "胶片颗粒",
        "shallow DOF": "浅景深",
        "bokeh": "背景虚化",
        "motion blur": "运动模糊",
        "color grading": "色彩分级"
    }
}

# ====== 工具函数 ======

def get_style(style_name: str) -> Optional[Dict]:
    """
    获取风格配置

    Args:
        style_name: 风格名称或标识

    Returns:
        风格配置字典，不存在则返回 None
    """
    # 模糊匹配
    for key, style in VIDEO_STYLES.items():
        if style_name.lower() in [key.lower(), style["name"].lower(), style["name_en"].lower()]:
            return style

    # 如果找不到，返回默认风格
    return VIDEO_STYLES.get(style_name.lower())

def get_all_styles() -> List[str]:
    """
    获取所有风格标识

    Returns:
        风格标识列表
    """
    return list(VIDEO_STYLES.keys())

def get_random_enhancement(category: str, count: int = 2) -> List[str]:
    """
    随机选择增强元素

    Args:
        category: 增强类别 (lighting, composition, atmosphere, technical)
        count: 选择数量

    Returns:
        选择的增强元素列表
    """
    if category not in ENHANCEMENTS:
        return []

    options = list(ENHANCEMENTS[category].keys())
    count = min(count, len(options))
    return random.sample(options, count)

def generate_prompt_enhancement(
    base_prompt: str,
    style: Dict,
    enhance: bool = True,
    language: str = "auto"
) -> str:
    """
    生成增强的提示词

    Args:
        base_prompt: 基础提示词
        style: 风格配置
        enhance: 是否增强提示词
        language: 语言 (en, zh, auto)

    Returns:
        增强后的提示词
    """
    if not enhance:
        return base_prompt

    # 获取风格特定的增强词
    style_enhancements = style.get("enhancements", {})

    # 随机选择各类增强词
    lighting = random.sample(style_enhancements.get("lighting", ["soft lighting"]), 1)[0]
    composition = random.sample(style_enhancements.get("composition", ["wide angle"]), 1)[0]
    mood = random.sample(style_enhancements.get("mood", ["cinematic"]), 1)[0]
    technical = random.sample(style_enhancements.get("technical", ["4K"]), 1)[0]

    # 构建增强提示词
    enhanced = f"{base_prompt}, {lighting}, {composition}, {mood}, {technical}"

    return enhanced

def generate_prompt_from_keywords(
    keywords: List[str],
    style: Optional[Dict] = None,
    enhance: bool = True
) -> str:
    """
    从关键词生成提示词

    Args:
        keywords: 关键词列表
        style: 风格配置（可选）
        enhance: 是否增强

    Returns:
        生成的提示词
    """
    base = ", ".join(keywords)

    if style:
        return generate_prompt_enhancement(base, style, enhance)

    return base

def generate_prompt_variants(
    topic: str,
    style: Dict,
    count: int = 3,
    enhance: bool = True
) -> List[str]:
    """
    生成多个提示词变体

    Args:
        topic: 主题
        style: 风格配置
        count: 生成数量
        enhance: 是否增强

    Returns:
        提示词列表
    """
    variants = []

    for _ in range(count):
        # 随机选择一些关键词
        style_keywords = style["keywords"]
        selected_keywords = random.sample(style_keywords, min(3, len(style_keywords)))

        # 构建基础提示词
        base_parts = [topic] + selected_keywords
        base_prompt = ", ".join(base_parts)

        # 增强提示词
        if enhance:
            enhanced_prompt = generate_prompt_enhancement(base_prompt, style, enhance)
            variants.append(enhanced_prompt)
        else:
            variants.append(base_prompt)

    return variants

def format_prompt_output(
    prompts: List[str],
    style: Dict,
    format_type: str = "readable"
) -> str:
    """
    格式化输出提示词

    Args:
        prompts: 提示词列表
        style: 风格配置
        format_type: 输出格式 (readable, json, markdown)

    Returns:
        格式化后的字符串
    """
    if format_type == "json":
        import json
        return json.dumps({
            "style": style["name"],
            "style_en": style["name_en"],
            "prompts": prompts
        }, ensure_ascii=False, indent=2)

    elif format_type == "markdown":
        output = f"# {style['name']} ({style['name_en']})\n\n"
        output += f"**描述**: {style['description']}\n\n"
        output += "## 生成的提示词\n\n"

        for i, prompt in enumerate(prompts, 1):
            output += f"### 提示词 {i}\n\n"
            output += f"```\n{prompt}\n```\n\n"

        return output

    else:  # readable
        output = f"\n🎨 风格: {style['name']} ({style['name_en']})\n"
        output += f"📝 描述: {style['description']}\n"
        output += f"{'='*60}\n\n"

        for i, prompt in enumerate(prompts, 1):
            output += f"✨ 提示词 {i}:\n"
            output += f"   {prompt}\n\n"

        return output

File: skills/video-prompt-generator/templates.js
/**
 * Video Prompt Generator - Prompt Templates
 *
 * Organized by category with 10+ video-specific creative styles
 * Each template includes placeholders for dynamic topic insertion
 */

const templates = {
  landscape: [
    {
      style: "Serene Mountain Sunrise",
      emoji: "🏔️",
      prompt: (topic) => `Cinematic video of ${topic} at sunrise over misty mountains, golden hour light breaking through clouds, slow camera movement revealing scenic panorama, dramatic lighting with soft shadows, 4K quality, peaceful and majestic atmosphere`
    },
    {
      style: "Ocean Sunset Drone Shot",
      emoji: "🌊",
      prompt: (topic) => `Aerial drone shot of ${topic} by the ocean at sunset, golden and orange hues reflecting on water, slow descent toward coastline, cinematic composition, romantic and peaceful mood, seagulls flying in background`
    },
    {
      style: "Forest Canopy Walkthrough",
      emoji: "🌲",
      prompt: (topic) => `First-person camera walking through ${topic} in lush forest, sunlight filtering through green canopy, dappled light patterns on path, immersive and serene atmosphere, gentle ambient sound, natural and organic feel`
    },
    {
      style: "Urban City Timelapse",
      emoji: "🏙",
      prompt: (topic) => `Timelapse video of ${topic} in modern city center, day-to-night transition, buildings lighting up, cars flowing like light rivers, dynamic energy, vibrant urban atmosphere, cyberpunk aesthetic`
    },
    {
      style: "Seasonal Cherry Blossom",
      emoji: "🌸",
      prompt: (topic) => `${topic} surrounded by falling cherry blossom petals, gentle spring breeze, soft pink and white flowers, romantic and dreamy atmosphere, slow motion capture of petals dancing in air`
    }
  ],
  product: [
    {
      style: "Cinematic Product Reveal",
      emoji: "📦",
      prompt: (topic) => `Slow cinematic reveal of ${topic} with dramatic lighting, rotating 360-degree showcase, sleek black background, premium product photography aesthetic, professional and sophisticated presentation`
    },
    {
      style: "Lifestyle Product Usage",
      emoji: "📱",
      prompt: (topic) => `${topic} in authentic lifestyle setting, person using product naturally, candid shot, warm natural lighting, relatable and aspirational atmosphere, product in focus with soft background blur`
    },
    {
      style: "E-commerce Flat Lay",
      emoji: "📦",
      prompt: (topic) => `Flat lay video of ${topic} on clean marble surface, subtle floating motion, minimal props, soft shadows, premium product photography, elegant and sophisticated aesthetic`
    },
    {
      style: "Exploded View Animation",
      emoji: "🔧",
      prompt: (topic) => `${topic} in exploded 3D view, components floating and assembling, clean white background, technical and informative, smooth animation, product understanding and assembly process`
    },
    {
      style: "Unboxing Experience",
      emoji: "📦",
      prompt: (topic) => `First-person unboxing experience of ${topic}, box opening with excitement, product reveal, packaging details, premium feel, celebratory and joyful atmosphere`
    }
  ],
  tech: [
    {
      style: "Cyberpunk Neon City",
      emoji: "🌃",
      prompt: (topic) => `${topic} in cyberpunk cityscape at night, neon lights reflecting on wet streets, holographic displays, flying cars in background, futuristic and edgy atmosphere, Blade Runner aesthetic`
    },
    {
      style: "AI Digital Interface",
      emoji: "🤖",
      prompt: (topic) => `${topic} with floating AI interface elements, holographic data visualizations, neural network patterns, futuristic tech aesthetic, blue and cyan lighting, innovative and cutting-edge`
    },
    {
      style: "Space Station View",
      emoji: "🚀",
      prompt: (topic) => `${topic} aboard space station, Earth visible through window, zero gravity floating, stars and nebula in background, awe-inspiring and cosmic, cinematic sci-fi aesthetic`
    },
    {
      style: "Digital Glitch Art",
      emoji: "📺",
      prompt: (topic) => `${topic} with digital glitch effects, pixelated distortions, RGB color splits, cyberpunk aesthetic, tech-forward design, edgy and contemporary`
    },
    {
      style: "Futuristic Laboratory",
      emoji: "🔬",
      prompt: (topic) => `${topic} in advanced laboratory setting, holographic displays, robotic arms, clean white and blue color scheme, innovative and scientific, high-tech research aesthetic`
    }
  ],
  emotional: [
    {
      style: "Romantic Moonlight Scene",
      emoji: "🌙",
      prompt: (topic) => `${topic} under moonlight, romantic and intimate, soft silver lighting, emotional connection, slow camera movements, love story atmosphere`
    },
    {
      style: "Nostalgic Vintage Film",
      emoji: "🎞",
      prompt: (topic) => `${topic} in vintage film aesthetic, sepia tones, warm nostalgic feeling, memory and reminiscence, classic and timeless`
    },
    {
      style: "Inspiring Journey",
      emoji: "🌟",
      prompt: (topic) => `${topic} representing journey and growth, dramatic landscape transition, hopeful music, motivational energy, rising from challenge to triumph`
    },
    {
      style: "Bittersweet Goodbye",
      emoji: "💔",
      prompt: (topic) => `${topic} in bittersweet farewell scene, emotional and touching, mix of sadness and hope, intimate close-up, authentic feeling`
    },
    {
      style: "Celebration and Joy",
      emoji: "🎉",
      prompt: (topic) => `${topic} in moment of pure celebration, confetti and joy, bright happy colors, genuine happiness and excitement`
    }
  ],
  urban: [
    {
      style: "Street Cafe Morning",
      emoji: "☕",
      prompt: (topic) => `${topic} in bustling street cafe scene, morning sunlight, people enjoying coffee, vibrant urban energy, authentic and relatable, cozy and welcoming atmosphere`
    },
    {
      style: "Modern Office Workspace",
      emoji: "💼",
      prompt: (topic) => `${topic} in stylish modern office, clean desk setup with plants, productivity and focus, natural lighting, contemporary work-from-home aesthetic`
    },
    {
      style: "Night City Walking",
      emoji: "🌃",
      prompt: (topic) => `${topic} walking through city at night, neon lights reflecting, urban exploration, cinematic composition, mysterious and alluring`
    },
    {
      style: "Subway Commute",
      emoji: "🚇",
      prompt: (topic) => `${topic} in subway commute scene, underground lighting, authentic daily life, movement and energy, urban realism`
    },
    {
      style: "Rooftop City View",
      emoji: "🌆",
      prompt: (topic) => `${topic} with city skyline view, golden hour sunset, urban lifestyle, aspirational and freeing, epic and cinematic`
    }
  ],
  food: [
    {
      style: "Food Preparation Close-up",
      emoji: "🍽",
      prompt: (topic) => `Close-up video of ${topic} preparation, fresh ingredients, sharp knife movements, appetizing food photography, mouth-watering detail, professional culinary aesthetic`
    },
    {
      style: "Cooking Process",
      emoji: "👨‍🍳",
      prompt: (topic) => `${topic} cooking in modern kitchen, steam rising, sizzling sounds, chef preparing with expertise, appetizing and inviting, culinary expertise showcase`
    },
    {
      style: "Plating Presentation",
      emoji: "🍴",
      prompt: (topic) => `Elegant plating of ${topic} on white plate, artistic arrangement, garnish details, fine dining aesthetic, professional and sophisticated`
    },
    {
      style: "Farm to Table",
      emoji: "🌱",
      prompt: (topic) => `${topic} from farm to table journey, fresh ingredients, natural setting, organic and sustainable, authentic and wholesome`
    },
    {
      style: "Food Porn Aesthetic",
      emoji: "📸",
      prompt: (topic) => `Slow-motion video of ${topic} with dramatic lighting, glistening textures, extreme close-up, food porn aesthetic, irresistible and appetizing`
    }
  ],
  sports: [
    {
      style: "Dynamic Sports Action",
      emoji: "⚽",
      prompt: (topic) => `${topic} in dynamic sports action, fast movement, intense energy, competitive spirit, slow-motion capture of key moments, athletic excellence`
    },
    {
      style: "Gym Workout Routine",
      emoji: "🏋",
      prompt: (topic) => `${topic} in modern gym setting, dedicated workout routine, fitness motivation, energy and determination, healthy lifestyle promotion`
    },
    {
      style: "Outdoor Adventure",
      emoji: "🏔",
      prompt: (topic) => `${topic} in outdoor adventure setting, nature background, active lifestyle, freedom and exploration, inspiring and energetic`
    },
    {
      style: "Yoga and Meditation",
      emoji: "🧘",
      prompt: (topic) => `${topic} in peaceful yoga or meditation scene, calm and centered, natural lighting, mindfulness and wellness, serene atmosphere`
    },
    {
      style: "Team Spirit",
      emoji: "🏆",
      prompt: (topic) => `${topic} capturing team spirit and camaraderie, group celebration, unity and collaboration, uplifting and motivational`
    }
  ],
  ancient: [
    {
      style: "Traditional Chinese Garden",
      emoji: "🏯",
      prompt: (topic) => `${topic} in traditional Chinese garden, ancient architecture, koi pond, cherry blossoms, serene and timeless, cultural authenticity`
    },
    {
      style: "Hanfu Costume Showcase",
      emoji: "👘",
      prompt: (topic) => `${topic} wearing elegant Hanfu costume, flowing fabric, traditional Chinese setting, cultural heritage, graceful and poetic`
    },
    {
      style: "Ink Wash Painting Style",
      emoji: "🖌️",
      prompt: (topic) => `${topic} in traditional Chinese ink wash painting aesthetic, brush stroke textures, monochromatic with color accents, artistic and cultural`
    },
    {
      style: "Ancient Architecture",
      emoji: "🏛️",
      prompt: (topic) => `${topic} with ancient Chinese architecture background, traditional patterns, cultural symbolism, historical and majestic`
    },
    {
      style: "Wuxia Martial Arts",
      emoji: "⚔",
      prompt: (topic) => `${topic} in wuxia martial arts scene, flowing movements, traditional Chinese setting, heroic and cinematic, ancient martial arts aesthetic`
    }
  ],
  anime: [
    {
      style: "Anime Opening Style",
      emoji: "🎬",
      prompt: (topic) => `${topic} in anime opening sequence style, vibrant colors, dynamic action poses, speed lines, anime aesthetic, energetic and stylized`
    },
    {
      style: "Kawaii Chibi Character",
      emoji: "😊",
      prompt: (topic) => `${topic} as cute kawaii chibi character, big expressive eyes, soft rounded shapes, adorable and appealing, playful and fun`
    },
    {
      style: "Fantasy School Life",
      emoji: "🏫",
      prompt: (topic) => `${topic} in anime fantasy school life setting, colorful classrooms, magical elements, slice of life with fantasy twist`
    },
    {
      style: "Mecha Robot Battle",
      emoji: "🤖",
      prompt: (topic) => `${topic} in epic mecha robot battle scene, glowing energy beams, dramatic explosions, sci-fi anime aesthetic, intense and spectacular`
    },
    {
      style: "Magical Girl Transformation",
      emoji: "✨",
      prompt: (topic) => `${topic} in magical girl transformation scene, sparkles and light effects, flowing ribbons, magical and whimsical, anime aesthetic`
    }
  ],
  abstract: [
    {
      style: "Liquid Abstract Art",
      emoji: "💧",
      prompt: (topic) => `${topic} morphing in liquid abstract art, flowing colors, organic shapes, dreamy and surreal, artistic and experimental`
    },
    {
      style: "Geometric Patterns",
      emoji: "🔷",
      prompt: (topic) => `${topic} with geometric pattern overlay, repeating shapes, clean lines, mathematical beauty, modern and minimalist`
    },
    {
      style: "Particle Explosion",
      emoji: "💥",
      prompt: (topic) => `${topic} with particle explosion effects, glowing points, burst of energy, dynamic and vibrant, futuristic and stunning`
    },
    {
      style: "Ink in Water",
      emoji: "🖌️",
      prompt: (topic) => `${topic} with ink dropping into water, swirling patterns, organic diffusion, artistic and meditative, black and white aesthetic`
    },
    {
      style: "Fractal Zoom",
      emoji: "🌀",
      prompt: (topic) => `${topic} zooming into fractal patterns, infinite detail, mathematical beauty, psychedelic and mesmerizing, trippy and abstract`
    }
  ]
};

// Get all categories (short names)
function getAllCategories() {
  return Object.keys(templates);
}

// Get templates by category
function getTemplatesByCategory(category) {
  return templates[category] || [];
}

// Get all templates
function getAllTemplates() {
  return templates;
}

// Generate prompt for specific category and style
function generatePrompt(category, styleIndex, topic) {
  const categoryTemplates = templates[category];
  if (!categoryTemplates || !categoryTemplates[styleIndex]) {
    throw new Error(`Invalid category or style index: ${category}, ${styleIndex}`);
  }

  const style = categoryTemplates[styleIndex];
  return {
    category: category,
    style: style.style,
    prompt: style.prompt(topic)
  };
}

// Generate all prompts for a topic
function generateAllPrompts(topic) {
  const results = [];

  for (const [category, styles] of Object.entries(templates)) {
    styles.forEach((style, index) => {
      results.push({
        category: category,
        style: style.style,
        prompt: style.prompt(topic)
      });
    });
  }

  return results;
}

// Generate prompts for specific categories
function generatePromptsForCategories(categories, topic) {
  const results = [];

  categories.forEach(category => {
    const categoryTemplates = templates[category];
    if (categoryTemplates) {
      categoryTemplates.forEach((style) => {
        results.push({
          category: category,
          style: style.style,
          prompt: style.prompt(topic)
        });
      });
    }
  });

  return results;
}

// Get count of total templates
function getTotalTemplateCount() {
  let count = 0;
  for (const category in templates) {
    count += templates[category].length;
  }
  return count;
}

module.exports = {
  templates,
  getAllCategories,
  getTemplatesByCategory,
  getAllTemplates,
  generatePrompt,
  generateAllPrompts,
  generatePromptsForCategories,
  getTotalTemplateCount
};
